Update for Elm 0.19

isaacseymour opened the pull request:
- Run elm-upgrade and elm-format
- Replace toString with String.fromInt
- Remove variable shadowing
- Change `to`/`fromTimestamp` to `to`/`fromPosix`
- Remove Debug.crash from time zone data generation (instead generate tests that ensure all time zones are validly parsed)

TODO:
- [x] Get the Iso8601ErrorMsg module working again
- [x] Remove Debug.todo from the ISO8601 parser's getIndex function (I think this could maybe be done with a nice zip kinda function instead?)
- [x] Remove Debug.todo from `TimeZone.find` (I think this could be done by changing the `List Span` type to a list of offsets, with implicit ±Infinity at either end)
- [x] Remove Debug.todo from `Date.unsafeDaysInMonth` (I think this just means making it safe, but not totally clear on that yet)

Questions:
- I added an internal `Month` type, to remove `unsafeDaysIntMonth`. Should that type be exposed to the world? It would be nicer imo, but would be a huge API change.

isaacseymour committed:
Update for Elm 0.19 (WIP)

- Run elm-upgrade and elm-format
- Remove Debug.crash from time zone data generation (instead generate
  tests that ensure all time zones are validly parsed)

isaacseymour committed:
Fix doctests for DateTime

isaacseymour committed:
Switch to advanced parser, add context and problems

isaacseymour committed:
Don't check in verify-examples

isaacseymour committed:
Remove Debug.todo from Date

isaacseymour committed:
Use Month type in Date module, to remove Debug.todo nicely

isaacseymour committed:
Convert remaining Debug.todos into parser failures

isaacseymour committed:
Fix up the error message stuff

isaacseymour committed:
format

isaacseymour committed:
Minimise work for elm-format

isaacseymour committed:
Update travis versions

isaacseymour committed:
Specify valid elm-test version

isaacseymour committed:
Update examples to 0.19 too

stephenreddek commented:
Thank you for working on this upgrade! I've run into a wall in upgrading without this package. I'll try to do a more thorough look at the changes, but quickly, I have a couple of questions:

1. Per this discussion: https://discourse.elm-lang.org/t/time-zone-explorations/1837/6 Would it make sense to add the lazy `() ->` back into the zone data?
2. Could you use `Time.Month` instead of a custom one?
3. Could you base the Iso 8601 parsing off of rtfeldman's package so his bug fixes could be applied easier? I believe this code was originally based off his parser and he's since made bug fixes.

Out of curiosity, how do you see this future of this package? It seems like the concepts of `ZonedDateTime` and `DateTime` could just be replaced with `Posix`. I personally find the most value in the support for timezone names, abbreviations, and, of course, the IANA pack.

isaacseymour commented:
There's been a discussion on Slack which looks like this package will move
out of elm-community and mostly be abandoned. I've updated this to unblock
my company's 0.19 upgrade. Once we've updated, I'll probably try to switch
to using Richard Feldman's package and justinmimbs/date. I'll publish this
under my own name with a clearer readme tomorrow's

On Wed, 5 Sep 2018, 20:16 Stephen Reddekopp, <notifications@github.com>
wrote:

> Thank you for working on this upgrade! I've run into a wall in upgrading
> without this package. I'll try to do a more thorough look at the changes,
> but quickly, I have a couple of questions:
>
>    1. Per this discussion:
>    https://discourse.elm-lang.org/t/time-zone-explorations/1837/6 Would
>    it make sense to add the lazy () -> back into the zone data?
>    2. Could you use Time.Month instead of a custom one?
>    3. Could you base the Iso 8601 parsing off of rtfeldman's package so
>    his bug fixes could be applied easier? I believe this code was originally
>    based off his parser and he's since made bug fixes.
>
> Out of curiosity, how do you see this future of this package? It seems
> like the concepts of ZonedDateTime and DateTime could just be replaced
> with Posix. I personally find the most value in the support for timezone
> names, abbreviations, and, of course, the IANA pack.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/elm-community/elm-time/pull/80#issuecomment-418849273>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABv8QZB3CiRinm31plyRoDOWfP9YFiRSks5uYCL8gaJpZM4WakZZ>
> .
>

isaacseymour commented:
Closing this PR. I'll be publishing my fork imminently.

isaacseymour commented:
Package now available as [isaacseymour/deprecated-time](https://package.elm-lang.org/packages/isaacseymour/deprecated-time/latest)

oldfartdeveloper commented:
Per discussion, this has been deployed outside of `elm-development` and this package has been deprecated.  For the **elm 0.19** version, see **[isaacseymour/deprecated-time](https://package.elm-lang.org/packages/isaacseymour/deprecated-time/latest)**.

Hence, this pull request is closed.
